src/main/default/classes/CoffeeMachineTest.cls

```diff
@@ -0,0 +1,169 @@
+@isTest
+class CoffeeMachineTest {
```

Comment:
In general the test class is good; but you should try to limit the scope of what is being tested by each method more by reducing the dependency on other methods in the sut. Several test cases now use methods of the sut to obtain the desired state in which the target method/function can be tested. This in turn means that the unit test rassumes those methods work properly; ideally you don't want to have such assumptions in your test methods.